apps/cli/src/domain/repository.ts

```diff
@@ -32,3 +34,66 @@ export const checkPreCommitConfig =
         "Pre-commit configuration is present in the repository root",
     });
   };
+
+export const checkTurboConfig =
+  (monorepoDir: string) =>
+  async (
+    dependencies: Pick<
+      Dependencies,
+      "dependencyVersionValidator" | "packageJsonReader" | "repositoryReader"
+    >,
+    { minVersions }: Config,
+  ): Promise<ValidationCheckResult> => {
+    const { dependencyVersionValidator, packageJsonReader, repositoryReader } =
+      dependencies;
+    const checkName = "Turbo Configuration";
+
+    const turboResult = repositoryReader.existsTurboConfig(monorepoDir);
+    if (turboResult.isErr()) {
+      return ok({
+        checkName,
+        errorMessage: turboResult.error.message,
+        isValid: false,
+      });
+    }
+
+    const dependenciesResult = await packageJsonReader.getDependencies(
+      monorepoDir,
+      "dev",
+    );
+    if (dependenciesResult.isErr()) {
+      return ok({
+        checkName,
+        errorMessage: dependenciesResult.error.message,
+        isValid: false,
+      });
+    }
+
+    const turboDependency = dependenciesResult.value.find(
+      ({ name }) => name === "turbo",
+    );
+    if (!turboDependency) {
+      return ok({
+        checkName,
+        errorMessage:
+          "Turbo dependency not found in devDependencies. Please add 'turbo' to your devDependencies.",
+        isValid: false,
+      });
+    }
+
+    return dependencyVersionValidator.isValid(
```

Comment:
IMHO `semver` can be used directly as we use `zod`, without wrapping it in an adapter

I don't think in the future we'll have setups that will require different validator strategy swapped at runtime for the same checks.